KAFKA-15270; [3/N] - Integration tests for simple consume & committed…

lianetm opened the pull request:
This introduces a PlaintextAsyncConsumer file as a means of selectively enabling tests while the new consumer is fully implemented and integrated into the PlaintextConsumer

lianetm committed:
KAFKA-15270; [3/N] - Integration tests for simple consume & committed offsets

lianetm commented:
In the newly added PlaintextAsyncConsumer, all disabled test have one of the following tags:
`@Disabled("Needs review for async consumer")` or `@Disabled("Not supported by async consumer yet")`

lianetm committed:
Allow empty group id

lianetm commented:
@philipnee note that this PR includes a minor fix to allow empty group id in the async consumer, which is what the current KafkaConsumer allows ([here](https://github.com/philipnee/kafka/blob/3546b2c755bc3e927a29fb6e1e1a702ffcffdc3d/clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java#L2502C9-L2502C9))
This fix ensures that both consumers keep the same behaviour at the API level (came out while running the existing integration test `testConsumingWithEmptyGroupId`

philipnee reviewed (approved):
Thanks, we will stop supporting empty group id based on this ticket: [KAFKA-14438](https://issues.apache.org/jira/browse/KAFKA-14438)

lianetm committed:
Revert "Allow empty group id"

This reverts commit 6ee4f4979f026a204b72b8ef21f559c4881cb49f.

lianetm committed:
Adding test for empty group not supported

lianetm committed:
Update comment on unsupported test

lianetm commented:
Thanks for the clarification regarding the empty group @philipnee. That introduces a change at the API level so I wasn't expecting it. Reverted the change and added a test for it, aligned with [KAFKA-14438](https://issues.apache.org/jira/browse/KAFKA-14438)
